R/hp_spacyr.R

```diff
@@ -1,15 +1,21 @@
-#' Render a Hierplane from Text using Spacy
-#' @param txt A sentence to be parsed and rendered as hierplane
-#' @param settings A hierplane settings object.
-#' @param ... Parameters to pass to `hierplane()`
+#' Create a hierplane object using spacyr
 #'
-#' @md
+#' Creating a hierplane object from a string is possible by using the `spacyr`
+#' package. More specifically, we rely on `spacyr::spacy_parse()` to tokenize and
+#' tag the string provided. Note that this functionality requires the `spacyr`
+#' package.
 #'
+#' @param .data A string of text.
+#' @md
 #' @export
-hierplane_spacy <- function(txt, settings = spacy_default(), ...) {
+hp_spacyr <- function(.data) {
   requireNamespaceQuietStop("spacyr")
-  x <- transform_logical(spacy_df(txt))
-  hierplane(.data = x, settings = settings, title = txt, ...)
+  x <- build_tree(
+    x = transform_logical(spacy_df(.data)),
+    title = .data,
+    settings = spacy_default()
+  )
+  structure(x, class = c("hierplane", "hierplane_spacyr", class(x)))
```

Comment:
I think the class should be more generic. Maybe `hierplane_tree`. And we can add a check at the beginning of `hierplane` to give a more concise error message. `hierplane` should only accept `hierplane_tree` objects.

Reply:
So you want to replace "hierplane" with "hierplane_tree" right? Not "hierplane_spacyr" -> "hierplane_tree"? If so, that's fine with me, we will just have to refer to these as hierplane tree objects, not hierplane objects.

Reply:
Right now both the widget and json objects are tagged as "heirplanes". We definitely would need some way to differentiate them, so in `hierplane` we look for a `hierplane_tree` object, and in `renderHierplane` we look for a `hierplane` object. 

Hope this makes sense!

Reply:
Hm, I am still confused. I don't think renderHierplane needs to recognize the class, it just works. The json objects have 3 classes:
* hierplane
* hierplane_spacyr/dataframe
* whatever the original class was

Reply:
`renderHierplane` only works with an output from `hierplane()` right? If you run `class(hierplane(df_hierplane))` now, it shows `hierplane` + `htmlwidget`, which means the the output from `hp_dataframe`/`hp_spacyr` have the same class as `hierplane`, which in my opinion is confusing. Also if we have more `hp_` functions, we don't want to have to look for each of the different `hierplane_` tags when checking for input object class in `hierplane`.

Reply:
I see, okay that makes sense, I can change the class to hierplane_tree